### ISSUE TITLE
axios -> fetch

### ISSUE THREAD

**Issue by guoyunhe** [guoyunhe/mui-app-template#3]
I had some trouble when do unit testing with axios and a real back-end API server (CORS).

`axios` is based on xhr, which is provided by jsdom in jest/vitest environment. There are a lot of issues with jsdom's xhr implementation.

fetch is natively supported by Node.js 18+. Older Node.js versions can use [whatwg-fetch](https://www.npmjs.com/package/whatwg-fetch) as a polyfill. jsdom doesn't provide any fetch implementation.

In conclusion, fetch is a more consist choice when you need to run React components in both browsers and unit test runners (jest/vitest).
